library/include/hipblas.h

```diff
@@ -434,7 +435,34 @@ typedef enum
     HIPBLAS_SIDE_BOTH = 143
 } hipblasSideMode_t;
 
-/*! \brief Indicates the precision width of data stored in a blas type. */
+#ifdef HIPBLAS_V2
+
+// Replacing use of hipblasDatatype_t with hipDataType which will be used in a future release.
+typedef hipDataType hipblasDatatype_t;
+
+#define HIPBLAS_R_16F HIP_R_16F
+#define HIPBLAS_R_32F HIP_R_32F
+#define HIPBLAS_R_64F HIP_R_64F
+#define HIPBLAS_C_16F HIP_C_16F
+#define HIPBLAS_C_32F HIP_C_32F
+#define HIPBLAS_C_64F HIP_C_64F
+#define HIPBLAS_R_8I HIP_R_8I
+#define HIPBLAS_R_8U HIP_R_8U
+#define HIPBLAS_R_32I HIP_R_32I
+#define HIPBLAS_R_32U HIP_R_32U
+#define HIPBLAS_C_8I HIP_C_8I
+#define HIPBLAS_C_8U HIP_C_8U
+#define HIPBLAS_C_32I HIP_C_32I
+#define HIPBLAS_C_32U HIP_C_32U
+#define HIPBLAS_R_16B HIP_R_16BF
+#define HIPBLAS_C_16B HIP_C_16BF
+#define HIPBLAS_DATATYPE_INVALID hipDataType(255)
+
+#else
+
+// clang-format off
+HIPBLAS_DEPRECATED_MSG("hipblasDatatype_t is deprecated, and it will be replaced completely by hipDataType in the future. Compiling with -DHIPBLAS_V2 will replace hipblasDatatype_t in the API with hipDataType.")
```

Comment:
I expect this gets reported a large many times as on the enum right? Maybe can make a bit more concise?

Reply:
Yeah when building hipBLAS it really makes it difficult to see compilation. I shortened it, but best to use HIPBLAS_NO_DEPRECATED_WARNINGS for devs.